src/main/java/bio/terra/metadata/FSEnumDir.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.metadata;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import java.util.List;
+
+public class FSEnumDir extends FSObjectBase {
```

Comment:
Yes, it is not the best, but the alternative bothered me also: You return this class, but before you can use its content member, you have to know/check its state. Was it retrieved with contents or not? So using the sub-classing provides that check; you simply don't have the method to retrieve the contents.

The other alternative is to collapse into one class and then maintain state in the FSDir class about whether it has valid contents. If getContents is called, but no contents was retrieved, then throw a "this code is broken" error.

My thought was that the compile-time check was better than the run-time check. But I am very open to other alternatives.

Reply:
Can we use `MetadataEnumeration<FSObjectBase>` ?

Reply:
Discussed the choices with Andrea. We agreed that the most straightforward and readable thing is to rename FSEnumDir to FSDir so it would always be able to represent contents. In addition add some state "isEnumerated" to guide the transformation into the FSObjectModel structure.